Source/Core/DolphinQt/Debugger/CodeViewWidget.cpp

```diff
@@ -393,26 +394,28 @@ void CodeViewWidget::CalculateBranchIndentation()
   // build a 2D lookup table representing the columns and rows the arrow could be drawn in
   // and try to place all branch arrows in it as far left as possible
   std::vector<bool> arrow_space_used(columns * rows, false);
-  const auto index = [&](u32 column, u32 row) { return column * rows + row; };
-  const u32 first_visible_addr = AddressForRow(0);
-  const u32 last_visible_addr = AddressForRow(static_cast<int>(rows - 1));
-  for (CodeViewBranch& branch : m_branches)
-  {
+  const auto index = [&](u32 column, u32 row) {
+    ASSERT(row <= rows);
+    ASSERT(column <= columns);
+    return column * rows + row;
+  };
+
+  const auto add_branch_arrow = [&](CodeViewBranch& branch, u32 first_addr, u32 first_row,
+                                    u32 last_addr) {
     const u32 arrow_src_addr = branch.src_addr;
     const u32 arrow_dst_addr = branch.is_link ? branch.src_addr : branch.dst_addr;
-    const u32 arrow_addr_lower = std::min(arrow_src_addr, arrow_dst_addr);
-    const u32 arrow_addr_higher = std::max(arrow_src_addr, arrow_dst_addr);
-    const bool is_visible =
-        last_visible_addr >= arrow_addr_lower || first_visible_addr <= arrow_addr_higher;
+    const auto [arrow_addr_lower, arrow_addr_upper] = std::minmax(arrow_src_addr, arrow_dst_addr);
+
+    const bool lower_in_range = first_addr <= arrow_addr_lower && arrow_addr_lower <= last_addr;
+    const bool upper_in_range = first_addr <= arrow_addr_upper && arrow_addr_upper <= last_addr;
+    const bool is_visible = lower_in_range || upper_in_range;
```

Comment:
I'd replace this with `is_visible = std::max(arrow_addr_lower, first_addr) <= std::min(arrow_addr_upper, last_addr)`; that way we properly handle cases where neither end is inside the view.

Reply:
Changed, but note that the way `m_branches` is currently calculated means that this doesn't come up in practice (it only contains branches where the source is visible; it doesn't contain ones where the destination is visible but not the source, and it doesn't contain ones where neither of them are visible).  That might be worth changing in the future, but the current implementation doesn't handle it.

Reply:
True! But if someone were to change this in the future so, say, all branches of the currently visible function(s) were checked, it'll be nice that this works already.